docs/content/guides/dagster/enriching-with-software-defined-assets.mdx

```diff
@@ -0,0 +1,407 @@
+
+This isn't the case with software-defined assets:
+
+- **Assets track their dependencies,** so you don't need to define mega-jobs just to keep track. This allows you to avoid interruptions in dependency graphs and eliminate the need for root input managers.
+
+- **Using software-defined assets usually means writing less code**. Specifying the inputs to a software-defined asset defines the assets it depends on. This means you don't need to use `@graph` and `@job` to wire dependencies between ops.
+
+  This approach reduces the number of times an asset's name appears in your codebase by half, making improving your codebase's scalability. Refer to the [IO manager-based example](#materialize-two-interdependent-tables-without-an-io-manager) below to see this in action.
+
+---
+
+## When is it appropriate to use software-defined assets?
+
+You should use software-defined assets when:
+
+- You’re using Dagster to produce or maintain assets, AND
+- You know what those assets will be before you launch any runs. Note that using software-defined assets in one job doesn’t mean they must be used in all your jobs.
+
+If your use case doesn't meet these criteria, you can still use graphs and ops.
+
+Still not sure? Check out these examples to see what's a good fit and what isn't:
+
+| Use case                                                                                                 | Good fit? | Explanation                                                      |
+| -------------------------------------------------------------------------------------------------------- | --------- | ---------------------------------------------------------------- |
+| Every day, drop and recreate the `users` table and the `user_recommender_model` model that depends on it | Yes       | Assets are known before a run and are being updated              |
+| Every hour, add a partition to the `events` table                                                        | Yes       | Assets are known before a run and are being updated              |
+| Clicking a button refreshes the `recommender` model                                                      | Yes       | Assets are known before a run and are being updated              |
+| Every day, send emails to a set of users                                                                 | No        | No assets are being updated                                      |
+| Every day, read a file of user IDs and change the value of a particular attribute for each user          | No        | The set of assets to update is not known before running the job. |
+| Every day, scan my warehouse for tables that haven't been used in months and delete them                 | No        | The set of assets to update is not known before running the job. |
+
+---
+
+## How do software-defined assets work with other Dagster concepts?
```

Comment:
@sryza This is what I wound up doing with what was previously the `Concept mapping` section. I also took the liberty of grouping concepts together to make the info progress the same way it does in the sidenav (for now) and be a bit easier to parse.

Reply:
I think breaking it up into these sections is really helpful. I still think that this should go below the migration examples. IMO the migration examples are the heart of this document and what I would be trying to get to as fast as possible if I was reading this.  Reading this stuff first would stress me out that there's all this stuff I need to think about to understand software-defined assets.

Reply:
Agreed!

I think I'm getting hung up on how including this as the last section will break the 'flow' the current structure has. Specifically, the first few headings are questions (ex: `Why use assets?`) and the actual example section deviates from it (`Upgrading jobs...`)

I know your original draft sorta went this direction (1How to upgrade to...`) but it wasn't phrased as a question. What if we went with `How do I upgrade my jobs?` instead of `Upgrading jobs to...`? 

That way, we can move the example content up and have it be more of the star, the way it should be. Thoughts?

Reply:
> I think I'm getting hung up on how including this as the last section will break the 'flow' the current structure has. Specifically, the first few headings are questions (ex: Why use assets?) and the actual example section deviates from it (Upgrading jobs...)

I hear you

> I know your original draft sorta went this direction (How to upgrade to...) but it wasn't phrased as a question. What if we went with How do I upgrade my jobs? instead of Upgrading jobs to...?

Sounds great - I think that's better than the heading in my original draft.

Reply:
Done and done.